stdlib/SparseArrays/src/sparsematrix.jl

```diff
@@ -2738,6 +2738,9 @@ function _spsetnz_setindex!(A::AbstractSparseMatrixCSC{Tv}, x::Tv,
     end
     return A
 end
+# ignore indexing into a Number by range
+_getrowinds(I::Number, range) = I
+_getrowinds(I::AbstractVector{<:Integer}, range) = I[range]
```

Comment:
Might as well preserve that huge `@inbounds` block through here.

```suggestion
_getrowinds(I::AbstractVector{<:Integer}, range) = @inbounds I[range]
```